go/vt/vtctl/grpcvtctldserver/query.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package grpcvtctldserver
+
+import (
+	"fmt"
+	"time"
+
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/protoutil"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+	"vitess.io/vitess/go/vt/vtctl/schematools"
+
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
+	"vitess.io/vitess/go/vt/proto/vttime"
+)
+
+// 2023-08-05 07:15:16
+const mysqlTimestampLayout = "2006-01-02 15:04:05"
```

Comment:
Better reuse `sqltypes.TimestampFormat` from `go/sqltypes/type.go`. See also `TimestampFormatPrecision3` and `TimestampFormatPrecision6`.